crates/core/src/models/transaction.rs

```diff
@@ -81,7 +80,14 @@ impl ConvertibleStarknetTransaction for StarknetTransaction {
 
         let hash: H256 = self.transaction_hash()?.into();
 
-        let nonce: U256 = self.nonce()?.into();
+        let nonce: U256 = match &self.0 {
+            Transaction::Invoke(invoke_tx) => match invoke_tx {
+                InvokeTransaction::V0(_) => Felt252Wrapper::ZERO,
```

Comment:
we can get the nonce here I think

https://github.com/kkrt-labs/kakarot-rpc/blob/02f32943f9e9afd6ab38be7ea17313ba2796a86c/crates/core/src/client/mod.rs#L339